website/src/scripts/setupTypedoc.ts

```diff
@@ -26,62 +26,67 @@ function updateMdFiles(file: string, sidebarInfo: string) {
-      fs.renameSync(readmeFile, path.resolve(dirname, "index.md"));
-    }
-
-    // remove the first two lines of navigator
-    const modulesFile = path.resolve(dirname, "modules.md");
+try {
+  const directories = fs.readdirSync(TYPEDOC_DIR);
+  directories.forEach((dir) => {
+    const dirname = path.resolve(TYPEDOC_DIR, dir);
+    const label = fitFormat(dir);
+
+    // only do things if it's a directory
+    if (fs.statSync(dirname).isDirectory()) {
+      const readmeFile = path.resolve(dirname, "README.md");
+
+      if (fs.existsSync(readmeFile)) {
+        updateMdFiles(readmeFile, generateSidebarString({ title: label, label }));
+        fs.renameSync(readmeFile, path.resolve(dirname, "index.md"));
+      }
 
-    if (fs.existsSync(modulesFile)) {
-      updateMdFiles(modulesFile, generateSidebarString({ title: `${label} Module`, label: "module", position: 1 }));
-    }
+      // remove the first two lines of navigator
+      const modulesFile = path.resolve(dirname, "modules.md");
 
-    const innerDirs = fs.readdirSync(dirname);
-    innerDirs.forEach((innerDir) => {
-      const innerDirname = path.resolve(dirname, innerDir);
-
-      if (fs.statSync(innerDirname).isDirectory()) {
-        const innerFiles = fs.readdirSync(innerDirname);
-        innerFiles.forEach((innerFile) => {
-          const innerLabel = innerFile.split(".")[0];
-          updateMdFiles(
-            path.resolve(innerDirname, innerFile),
-            generateSidebarString({ title: innerLabel, label: innerLabel }),
-          );
-        });
+      if (fs.existsSync(modulesFile)) {
+        updateMdFiles(modulesFile, generateSidebarString({ title: `${label} Module`, label: "module", position: 1 }));
       }
-    });
+
+      const innerDirs = fs.readdirSync(dirname);
+      innerDirs.forEach((innerDir) => {
+        const innerDirname = path.resolve(dirname, innerDir);
+
+        if (fs.statSync(innerDirname).isDirectory()) {
+          const innerFiles = fs.readdirSync(innerDirname);
+          innerFiles.forEach((innerFile) => {
+            const innerLabel = innerFile.split(".")[0];
+            updateMdFiles(
+              path.resolve(innerDirname, innerFile),
+              generateSidebarString({ title: innerLabel, label: innerLabel }),
+            );
+          });
+        }
+      });
+    }
+  });
+
+  // insert index page
+  insertIndexPage(TYPEDOC_DIR, { title: "Typedoc", label: "Typedoc" });
+
+  // find the target moving directory
+  const versionFile = path.resolve(__dirname, "../../versions.json");
+  let versionDir = "";
+  try {
+    const versionContent = fs.readFileSync(versionFile, "utf8");
+    if (versionContent) {
+      const versionContentJson = JSON.parse(versionContent) as string[];
+      versionDir = path.resolve(__dirname, `../../versioned_docs/version-${versionContentJson[0]}/typedoc`);
+    }
+  } catch (e) {
+    versionDir = path.resolve(__dirname, "../../docs/typedoc");
   }
```

Comment:
Nested try/catch is not a good practice as well as using error handing for logic handling.

Reply:
How about removing try...catch by check if file exists before?
But still nested if...else

Or would you suggest to split them to several functions?

Reply:
I think yes, it'd be better to check this logic with if/else and use functions.